core/src/main/java/org/infinispan/cache/impl/EncoderCache.java

```diff
@@ -0,0 +1,878 @@
+   }
+
+   @Override
+   public CompletableFuture<Boolean> replaceAsync(K key, V oldValue, V newValue, long lifespan, TimeUnit unit) {
+      return super.replaceAsync(keyToStorage(key), valueToStorage(oldValue), valueToStorage(newValue), lifespan, unit);
+   }
+
+   @Override
+   public V put(K key, V value, Metadata metadata) {
+      V ret = super.put(keyToStorage(key), valueToStorage(value), metadata);
+      return valueFromStorage(ret);
+   }
+
+   @Override
+   public V replace(K key, V value, Metadata metadata) {
+      V ret = super.replace(keyToStorage(key), valueToStorage(value), metadata);
+      return valueFromStorage(ret);
+   }
+
+   @Override
+   public CompletableFuture<Boolean> replaceAsync(K key, V oldValue, V newValue, long lifespan, TimeUnit lifespanUnit, long maxIdle, TimeUnit maxIdleUnit) {
+      return super.replaceAsync(keyToStorage(key), valueToStorage(oldValue), valueToStorage(newValue), lifespan, lifespanUnit, maxIdle, maxIdleUnit);
+   }
+
+   @Override
+   public boolean replace(K key, V oldValue, V value, Metadata metadata) {
+      return super.replace(keyToStorage(key), valueToStorage(oldValue), valueToStorage(value), metadata);
+   }
+
+   @Override
+   public V putIfAbsent(K key, V value, Metadata metadata) {
+      V ret = super.putIfAbsent(keyToStorage(key), valueToStorage(value), metadata);
+      return valueFromStorage(ret);
+   }
+
+   @Override
+   public CompletableFuture<V> putAsync(K key, V value, Metadata metadata) {
+      return super.putAsync(keyToStorage(key), valueToStorage(value), metadata).thenApply(transcodeValueForRead);
+   }
+
+   @Override
+   public void putForExternalRead(K key, V value, Metadata metadata) {
+      super.putForExternalRead(keyToStorage(key), valueToStorage(value), metadata);
+   }
+
+   @Override
+   public void putAll(Map<? extends K, ? extends V> map, Metadata metadata) {
+      super.putAll(transcodeMapForWrite(map), metadata);
+   }
+
+   @Override
+   public CacheSet<CacheEntry<K, V>> cacheEntrySet() {
+      return new EncoderEntrySet(this, super.cacheEntrySet());
+   }
+
+   @Override
+   public void removeExpired(K key, V value, Long lifespan) {
+      super.removeExpired(keyToStorage(key), valueToStorage(value), lifespan);
+   }
+
+   @Override
+   public V putIfAbsent(K key, V value) {
+      V ret = super.putIfAbsent(keyToStorage(key), valueToStorage(value));
+      return valueFromStorage(ret);
+   }
+
+   @Override
+   public AdvancedCache<K, V> withEncoder(Encoder encoder) {
+      EncoderCache<K, V> encoderCache = new EncoderCache<>(cache);
+      encoderCache.entryFactory = this.entryFactory;
+      encoderCache.wrapper = this.wrapper;
+      encoderCache.encoder = encoder;
+      return encoderCache;
+   }
+
+   @Override
+   public AdvancedCache<K, V> withWrapper(Wrapper wrapper) {
+      EncoderCache<K, V> encoderCache = new EncoderCache<>(cache);
+      encoderCache.entryFactory = this.entryFactory;
```

Comment:
Same thing about the variables being set. Like `encodingType` seems to have been missed here.